src/qml/qgismobileapp.qml

```diff
@@ -861,6 +862,82 @@ ApplicationWindow {
     }
   }
 
+  Geofencer {
+    id: geofencer
+
+    position: positionSource.projectedPosition
+    positionCrs: mapCanvas.mapSettings.destinationCrs
+
+    onIsWithinChanged: {
+      if (behavior == Geofencer.AlertWhenInsideGeofencedArea && geofencer.isWithin) {
+        platformUtilities.vibrate(500)
+        displayToast(qsTr("Position has trespassed into ‘%1’").arg(isWithinAreaName), 'error')
+      } else if (behavior == Geofencer.AlertWhenOutsideGeofencedArea && !geofencer.isWithin) {
+        platformUtilities.vibrate(500)
+        displayToast(qsTr("Position outside areas after leaving ‘%1’").arg(lastWithinAreaName), 'error')
+      } else if (behavior == Geofencer.InformWhenEnteringLeavingGeofencedArea) {
+        if (isWithin) {
+          platformUtilities.vibrate(250)
+          displayToast(qsTr("Position entered into ‘%1’").arg(isWithinAreaName))
+        } else if (lastWithinAreaName != '') {
+          platformUtilities.vibrate(250)
+          displayToast(qsTr("Position left from ‘%1’").arg(lastWithinAreaName))
+        }
+      }
+    }
+  }
+
+  MultiEffect {
+    id: geofencerFeedback
+    anchors.fill: geofencerFeedbackSource
+    source: geofencerFeedbackSource
+    visible: true
+    blurEnabled: true
+    blurMax: 64
+    blur: 2.0
+    opacity: 0
+
+    SequentialAnimation {
+      id: geofencerFeedbackAnimation
+      running: geofencer.isAlerting
+      loops: Animation.Infinite
+
+      onRunningChanged: {
+        if (!running) {
+          geofencerFeedback.opacity = 0
+        }
+      }
+
+      OpacityAnimator {
+        target: geofencerFeedback
+        from: 0;
+        to: 0.75;
+        duration: 1000
+      }
+      OpacityAnimator {
+        target: geofencerFeedback
+        from: 0.75;
+        to: 0;
+        duration: 1000
+      }
+    }
+  }
+
+  Rectangle {
+    id: geofencerFeedbackSource
+    width: Math.min(250, mainWindow.width / 2)
+    height: width
+    radius: width / 2
+    visible: false
+
+    x: parent.width - width / 2
+    y: locationToolbar.y + gnssButton.y + (gnssButton.height / 2) - height / 2
+
+    color: Theme.errorColor
+    border.width: 2
+    border.color: "#ffffff"
```

Comment:
I'm not sure, can we use Theme.color ?

Reply:
Good catch, this was only use to tweak location, removed.